src/ItemsPerPageSelector.php

```diff
@@ -60,7 +60,7 @@ public function onPageLengthSelect(\Closure $fx): void
         $this->cb->set(function () use ($fx) {
             $ipp = isset($_GET['ipp']) ? (int) $_GET['ipp'] : null;
             // $this->pageLength->set(preg_replace("/\[ipp\]/", $ipp, $this->label));
-            $this->set($ipp);
+            $this->set((string) $ipp);
             $reload = $fx($ipp);
```

Comment:
In `init` method a few lines above there is the same cast. The `set` is there for setting `View::$content` prop probably, the best way would be probably to remove every `set` call in order to set it for the render in `renderView`.

This and whole menu items must be typecasted using UI persistence in order for format numbers > 1k / page.